ui/app/AppSearch.qml

```diff
@@ -11,18 +11,24 @@ Item {
     readonly property var searchMessages: Backpressure.debounce(searchPopup, 400, function (value) {
         appSearch.store.searchMessages(value)
     })
+    property alias opened: searchPopup.opened
 
     function openSearchPopup(){
         searchPopup.open()
     }
 
+    function closeSearchPopup(){
+        searchPopup.close()
+    }
+
     Connections {
         target: appSearch.store.locationMenuModel
         onModelAboutToBeReset: {
-            for (var i = 2; i <= searchPopupMenu.count; i++) {
-                //clear menu
-                if (!!searchPopupMenu.takeItem(i)) {
-                    searchPopupMenu.removeItem(searchPopupMenu.takeItem(i));
+            while (true) {
+                // Delete the item right after the default items
+                if (!searchPopupMenu.takeItem(searchPopupMenu.NUM_DEFAULT_ITEMS)) {
+                    // If takeItem returns null, it means there was nothing to remove, we can break
+                    break
                 }
             }
```

Comment:
```suggestion
            while (searchPopupMenu.takeItem(searchPopupMenu.NUM_DEFAULT_ITEMS)) {
                // Delete the item right after the default items
                // If takeItem returns null, it means there was nothing to remove
            }
```

Reply:
Could be moved to `StatusSearchLocationMenu.qml` ?
`onModelAboutToBeReset: searchPopupMenu.removeCustomItems()`

Btw. it feels wrong to remove items manually :thinking:  but it's for another discussion

Reply:
> Could it be moved to StatusSearchLocationMenu.qml ?

This was @saledjenic 's code. What do you think @saledjenic, can the reset code be moved to `StatusSearchLocationMenu.qml`?

Reply:
@jrainville in real that's @alexandraB99 code :) (that qml part). 

I am thinking now... and I am not sure why would we move that connection to `StatusSearchLocationMenu` component since that is actually a use case defined in our app, how we want to use `StatusSearchLocationMenu` component which is part of the `StatusQ`. Moving it there we will somehow limit it, or we can add another flag for for example to enable/disable that connection, but that just inflates a component.

All in all, I wouldn't move it, but also if you two decide to move it I am ok with that as well.

Reply:
I didn't mean to move the connection but only the part related to deleting custom items. Since `NUM_DEFAULT_ITEMS` is already in `StatusSearchLocationMenu` component it would make sense to also have `removeCustomItems()` function there. 
```
function removeCustomItems {
    while (searchPopupMenu.takeItem(searchPopupMenu.NUM_DEFAULT_ITEMS)) {}
}
```

Reply:
@osmaczko yes, that sounds reasonable.